Framework/WorkflowAlgorithms/test/ConvolutionFitSequentialTest.h

```diff
@@ -193,10 +196,81 @@ class ConvolutionFitSequentialTest : public CxxTest::TestSuite {
     TS_ASSERT_EQUALS(memberLogs.at(5)->value(), "ConvFit");
     TS_ASSERT_EQUALS(memberLogs.at(6)->value(), "ReductionWs_");
     TS_ASSERT_EQUALS(memberLogs.at(7)->value(), "1");
+
+    AnalysisDataService::Instance().clear();
+  }
+
+  void test_exec_with_sqw_file() {
+    auto sqwWs = createGenericWorkspace("SqwWs_", true);
+    auto resWs = createGenericWorkspace("ResolutionWs_", false);
+    auto convFitRes = createGenericWorkspace("__ConvFit_Resolution", false);
+    Mantid::Algorithms::ConvolutionFitSequential alg;
+    TS_ASSERT_THROWS_NOTHING(alg.initialize());
+    alg.setProperty("InputWorkspace", sqwWs);
+    alg.setProperty("Function",
+                    "name=LinearBackground,A0=0,A1=0,ties=(A0=0.000000,A1=0.0);"
+                    "(composite=Convolution,FixResolution=true,NumDeriv=true;"
+                    "name=Resolution,Workspace=__ConvFit_Resolution,"
+                    "WorkspaceIndex=0;((composite=ProductFunction,NumDeriv="
+                    "false;name=Lorentzian,Amplitude=1,PeakCentre=0,FWHM=0."
+                    "0175)))");
+    alg.setProperty("BackgroundType", "Fixed Flat");
+    alg.setProperty("StartX", 0.0);
+    alg.setProperty("EndX", 5.0);
+    alg.setProperty("SpecMin", 0);
+    alg.setProperty("SpecMax", 0);
+    alg.setProperty("Convolve", true);
+    alg.setProperty("Minimizer", "Levenberg-Marquardt");
+    alg.setProperty("MaxIterations", 500);
+    TS_ASSERT_THROWS_NOTHING(alg.execute());
+    TS_ASSERT(alg.isExecuted());
+
+    // Assert that output is in ADS
+    TS_ASSERT_THROWS_NOTHING(
+        AnalysisDataService::Instance().retrieveWS<ITableWorkspace>(
+            "SqwWs_conv_1LFixF_s0_to_0_Parameters"));
+
+    TS_ASSERT_THROWS_NOTHING(
+        AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>(
+            "SqwWs_conv_1LFixF_s0_to_0_Result"));
+
+    TS_ASSERT_THROWS_NOTHING(
+        AnalysisDataService::Instance().retrieveWS<WorkspaceGroup>(
+            "SqwWs_conv_1LFixF_s0_to_0_Workspaces"));
+
+    AnalysisDataService::Instance().clear();
   }
 
   //------------------------ Private Functions---------------------------
 
+  MatrixWorkspace_sptr createGenericWorkspace(const std::string &wsName,
+                                              const bool numericAxis) {
+    const auto xData = boost::assign::list_of(1)(2)(3)(4)(5)
```

Comment:
This is a good use of `boost::assign` but these days we can just use an `initializer_list`, e.g.
```
const std::vector<double> xData{1,2,3,4,5};
const std::vector<double> yData{0,1,3,1,0};
```